tools/kagami/src/crypto.rs

```diff
@@ -111,4 +243,53 @@ mod tests {
             format!("{}", Algorithm::Ed25519)
         )
     }
+
+    const GENESIS_JSON_PATH: &str = "../../configs/swarm/genesis.json";
+    const GEN_KEYPAIR_JSON_PATH: &str = "test_keypair_path_for_crypt0_genesis_kagami.json";
+    const GEN_SIGNED_ENCODED_GENESIS_PATH: &str =
+        "test_signed_encoded_genesis_path_for_crypt0_genesis_kagami";
+
+    fn genesis_signing_works() -> Result<()> {
+        let keypair_config = KeyPairArgs {
+            algorithm: AlgorithmArg::default(),
+            private_key: None,
+            seed: None,
+            json: true,
+            compact: false,
+        };
+
+        let mut keypair_json = BufWriter::new(Vec::new());
+        keypair_config.run(&mut keypair_json)?;
+
+        fs::write(GEN_KEYPAIR_JSON_PATH, keypair_json.buffer())?;
+
+        let crypto_genesis_config = GenesisSigningArgs {
+            algorithm: AlgorithmArg::default(),
+            private_key_string: None,
+            private_key_path: None,
+            public_key_string: None,
+            public_key_path: None,
+            keypair_path: Some(PathBuf::from_str(GEN_KEYPAIR_JSON_PATH)?),
+            chain_id: ChainId::from("0123456"),
+            genesis_path: PathBuf::from_str(GENESIS_JSON_PATH)?,
+            output_path: Some(PathBuf::from_str(GEN_SIGNED_ENCODED_GENESIS_PATH)?),
+        };
+
+        let mut genesis_buf_writer = BufWriter::new(Vec::new());
+
+        crypto_genesis_config.run(&mut genesis_buf_writer)?;
+
+        let signed_genesis_from_file_encoded = fs::read(GEN_SIGNED_ENCODED_GENESIS_PATH)?;
+        let _maybe_signed_genesis_from_file: GenesisNetwork =
```

Comment:
Would be cool to `assert_eq!` the resulting `GenesisNetwork` with an expected one. Shouldn't take many lines for this.

Reply:
I've tried to make it work. It does not due to different creation times thus payload is different.
Do you have something in mind on how to bypass this?

Reply:
Oh, that's unfortunate that there is this side effect sneaking deeply inside.

IMO this usage of `current_time()` should be moved away from depths of data model, to some peripheral layers of code:

https://github.com/hyperledger/iroha/blob/96bf9973c24eb7789c5389fc66890fb2f277d563/data_model/src/transaction.rs#L641

Or add a way to mock what `current_time` in tests?